server/src/main/java/com/google/musicanalysis/site/YoutubeGenres.java

```diff
@@ -68,12 +75,17 @@ protected void calculateMusicCount(JsonArray videos) {
             }
         }
 
-        if (isMusic && totalSubgenres == 0) {
+        if (isMusic) {
+          // likedMusicHistory records video numbers that are music
+          likedMusicHistory.add(firstVideoCount + i);
+
+          if (totalSubgenres == 0) {
             // video only classified as Music so we update as "Other music"
-            this.updateGenre("Other music");
+            this.updateGenre("Other music");            
+          }
         }
     }
-    return;
+    return videosSize;
```

Comment:
You can get this info from the parent function. No need to return it here, and that in fact obscures what this function is doing, since it analyzes the videos but also returns the total number. Someone interacting with this function will likely write:
```
totalVideos += calculateMusicVideos(videos, totalVideos);
```

This means 2 independent things are happening on one line that aren't strictly tied together. More clear is:
```
calculateMusicVideos(videos, totalVideos);
totalVideos += videos.size();
```